awx/ui_next/src/screens/ExecutionEnvironment/shared/ExecutionEnvironmentForm.jsx

```diff
@@ -37,19 +47,58 @@ function ExecutionEnvironmentFormFields({ i18n, me, executionEnvironment }) {
     [setFieldValue]
   );
 
+  const [
+    containerOptionsField,
+    containerOptionsMeta,
+    containerOptionsHelpers,
+  ] = useField({
+    name: 'pull',
+  });
+
+  const containerPullChoices = options?.actions?.POST?.pull?.choices.map(
+    ([value, label]) => ({ value, label, key: value })
+  );
+
   return (
     <>
+      <FormField
+        id="execution-environment-name"
+        label={i18n._(t`Name`)}
+        name="name"
+        type="text"
+        validate={required(null, i18n)}
+        isRequired
+      />
       <FormField
         id="execution-environment-image"
         label={i18n._(t`Image name`)}
         name="image"
         type="text"
-        validate={url(i18n)}
+        validate={required(null, i18n)}
         isRequired
         tooltip={i18n._(
           t`The registry location where the container is stored.`
         )}
       />
+      <FormGroup
+        fieldId="execution-environment-container-options"
+        helperTextInvalid={containerOptionsMeta.error}
+        validated={
+          !containerOptionsMeta.touched || !containerOptionsMeta.error
+            ? 'default'
+            : 'error'
+        }
+        label={i18n._(t`Container Pull Option`)}
```

Comment:
```suggestion
        label={i18n._(t`Pull`)}
```

(To make it consistent w/ the Details panel)